lib/Doctrine/ODM/MongoDB/Mapping/Driver/XmlDriver.php

```
@@ -335,15 +338,34 @@ public function loadMetadataForClass($className, \Doctrine\Persistence\Mapping\C
             }
         }
 
-        if (! isset($xmlRoot->{'also-load-methods'})) {
-            return;
+        if (isset($xmlRoot->{'also-load-methods'})) {
+            foreach ($xmlRoot->{'also-load-methods'}->{'also-load-method'} as $alsoLoadMethod) {
+                $metadata->registerAlsoLoadMethod((string) $alsoLoadMethod['method'], (string) $alsoLoadMethod['field']);
+            }
         }
 
-        foreach ($xmlRoot->{'also-load-methods'}->{'also-load-method'} as $alsoLoadMethod) {
-            $metadata->registerAlsoLoadMethod((string) $alsoLoadMethod['method'], (string) $alsoLoadMethod['field']);
+        if (isset($xmlRoot->{'time-series'})) {
+            $attributes = $xmlRoot->{'time-series'}->attributes();
+
+            $metaField             = isset($attributes['meta-field']) ? (string) $attributes['meta-field'] : null;
+            $granularity           = isset($attributes['granularity']) ? Granularity::from((string) $attributes['granularity']) : null;
```

Do you want to rely on [`BackedEnum::from()`](https://www.php.net/manual/en/backedenum.from.php) throwing a ValueError for a bad string value? I don't think we can trust that the XSD will be enforced.

An explicit error is probably better than silently accepting `null` via [`tryFrom()`](https://www.php.net/manual/en/backedenum.tryfrom.php).

---

While we do use an enumeration in the schema file, I don't quite want to rely on that. Using `BackedEnum::from` is the best way to ensure validity of the value while avoiding duplication.